### PR TITLE
Escape ] in rewritten xrefs for downstream doc

### PR DESCRIPTION
**Pull request by gsmet**
Supersedes #41651

### CODE REVIEW

File: docs/src/main/java/io/quarkus/docs/generation/AssembleDownstreamDocumentation.java [hunk 1]
```diff
@@ -466,6 +466,10 @@ private static String rewriteContent(String fileName,
         return content;
     }
 
+    private static String escapeXrefTitleForReplaceAll(String title) {
+        return title.trim().replace("]", "\\\\]");
```
Comment (gsmet) [inline docs/src/main/java/io/quarkus/docs/generation/AssembleDownstreamDocumentation.java:470]: And yes, you need that many backslashes :]

**Review by jmartisk — APPROVED** [review]

**Comment by gsmet** [comment]
I pushed this fix to 3.8 too.

File: docs/src/main/java/io/quarkus/docs/generation/AssembleDownstreamDocumentation.java (hunk 1 above)
Comment (MichalMaler) [reply docs/src/main/java/io/quarkus/docs/generation/AssembleDownstreamDocumentation.java:470]: Great job! :) Backslash magic.

**Comment by rolfedh** [comment]
Thank you, @gsmet! Very appreciated!

**Pull request merged**